### PR TITLE
Add invoice date to material invoices

### PR DESCRIPTION
**Pull request by Aditha-Sansa**

### CODE REVIEW

File: database/migrations/2022_08_23_071614_add_invoiced_date_column_to_material_invoices_table.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('material_invoices', function (Blueprint $table) {
```
Comment (nuke99) [inline database/migrations/2022_08_23_071614_add_invoiced_date_column_to_material_invoices_table.php:17]: put a condition to run this only on production.

File: database/migrations/2022_08_23_071614_add_invoiced_date_column_to_material_invoices_table.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            $table->dateTimeTz('invoiced_date')->after('updated_at')->nullable();
+        });
+
+        $modelCollection = \App\Models\MaterialInvoice::all();
```
Comment (nuke99) [inline database/migrations/2022_08_23_071614_add_invoiced_date_column_to_material_invoices_table.php:21]: This should be inside a job which can be dispatched from the migration

File: database/migrations/2022_08_23_084451_change_invoiced_date_column_to_non_nullable_material_invoices_table.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
```
Comment (nuke99) [inline database/migrations/2022_08_23_084451_change_invoiced_date_column_to_non_nullable_material_invoices_table.php:14]: why is this needed? Material doesn't need an invoiced date, right ?

File: resources/js/Pages/Invoices/InvoiceShow.vue [hunk 4]
```diff
@@ -149,6 +157,9 @@ export default {
             required: true
         }
     },
+    mounted() {
+        console.log(this.formatTime(this.invoice[0].created_at))
```
Comment (nuke99) [inline resources/js/Pages/Invoices/InvoiceShow.vue:161]: remove the console.log

**Review by nuke99 — CHANGES_REQUESTED** [review]

File: app/Jobs/FillInvoicedDate.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $modelCollection = \App\Models\MaterialInvoice::all();
```
Comment (nuke99) [inline app/Jobs/FillInvoicedDate.php:33]: only when invoice date is empty

Comment (Aditha-Sansa) [reply app/Jobs/FillInvoicedDate.php:33]: This is fixed

**Pull request merged**